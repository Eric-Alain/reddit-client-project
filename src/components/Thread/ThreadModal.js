//React
import React from 'react'

//3rd party
import PropTypes from "prop-types"
import Img from 'react-cool-img'
import Modal from 'react-bootstrap/Modal'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'

const ThreadModal = ({ show, onHide, img}) => {  
  return (
    <Modal show={show} onHide={onHide} dialogClassName='thread-modal' keyboard={false}>
      <Modal.Header className='justify-content-end mx-3 py-1 px-0'>
        <button className="lh-0 rounded" onClick={onHide}>
          <FontAwesomeIcon className='modal-close' icon={['fas', 'xmark']} size='2x' />
        </button>
      </Modal.Header>
      <Modal.Body className="pt-0">
        <Img className='w-100 rounded' src={img} alt={''} />
      </Modal.Body>
    </Modal>
  )
}

ThreadModal.propTypes = {
  show: PropTypes.bool.isRequired,
  onHide: PropTypes.func.isRequired,
  img: PropTypes.string.isRequired
}

export default ThreadModal