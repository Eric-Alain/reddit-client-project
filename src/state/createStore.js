//Redux
import { createStore as reduxCreateStore, combineReducers } from 'redux'
import { composeWithDevTools } from 'redux-devtools-extension'

//Reducers
import searchReducer from '../state/reducers/search'
import dataReducer from '../state/reducers/data'
import subredditsReducer from '../state/reducers/subreddits'
import togglesReducer from '../state/reducers/toggles'

//Create root reducer
export const rootReducer = combineReducers({
  search: searchReducer,
  data: dataReducer,
  subreddits: subredditsReducer,
  toggles: togglesReducer
})

//Create store, pass preloadedState via plugin
const createStore = preloadedState => {
  return reduxCreateStore(rootReducer, preloadedState, composeWithDevTools())
}
export default createStore
