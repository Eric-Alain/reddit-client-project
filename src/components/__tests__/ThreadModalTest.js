import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import ThreadModal from '../Thread/ThreadModal'

import { Provider } from 'react-redux'
import createStore from '../../state/createStore'
import registerIcons from '../../icons/icons'

(async () => {
  if (!('IntersectionObserver' in window)) await import('intersection-observer')
})()

const store = createStore()
registerIcons()

beforeAll(() => {
  ReactDOM.createPortal = jest.fn((element, node) => {
    return element
  })
})

describe('ThreadModal', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <ThreadModal show={true} onHide={jest.fn()} img={'http://placekitten.com/g/480/320'} />
        </Provider>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
