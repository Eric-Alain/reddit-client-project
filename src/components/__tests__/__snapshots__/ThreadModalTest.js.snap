// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ThreadModal renders correctly 1`] = `
Array [
  <div
    className="fade modal-backdrop show"
    onClick={[Function]}
  />,
  <div
    aria-modal={true}
    className="fade modal show"
    onClick={[Function]}
    onMouseUp={[Function]}
    role="dialog"
    style={Object {}}
    tabIndex={-1}
  >
    <div
      className="modal-dialog thread-modal"
      onMouseDown={[Function]}
    >
      <div
        className="modal-content"
      >
        <div
          className="justify-content-end mx-3 py-1 px-0 modal-header"
        >
          <button
            className="lh-0 rounded"
            onClick={[MockFunction]}
          >
            <svg
              aria-hidden="true"
              className="svg-inline--fa fa-xmark fa-2x modal-close"
              data-icon="xmark"
              data-prefix="fas"
              focusable="false"
              role="img"
              style={Object {}}
              viewBox="0 0 320 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M310.6 361.4c12.5 12.5 12.5 32.75 0 45.25C304.4 412.9 296.2 416 288 416s-16.38-3.125-22.62-9.375L160 301.3L54.63 406.6C48.38 412.9 40.19 416 32 416S15.63 412.9 9.375 406.6c-12.5-12.5-12.5-32.75 0-45.25l105.4-105.4L9.375 150.6c-12.5-12.5-12.5-32.75 0-45.25s32.75-12.5 45.25 0L160 210.8l105.4-105.4c12.5-12.5 32.75-12.5 45.25 0s12.5 32.75 0 45.25l-105.4 105.4L310.6 361.4z"
                fill="currentColor"
                style={Object {}}
              />
            </svg>
          </button>
        </div>
        <div
          className="pt-0 modal-body"
        >
          <img
            alt=""
            className="w-100 rounded no-js-320"
            src="data:image/gif;base64,R0lGODlhAQABAIAAAP///wAAACH5BAEAAAAALAAAAAABAAEAAAICRAEAOw=="
          />
          <noscript>
            <style>
              .no-js-320 { display: none !important; }
            </style>
            <img
              alt=""
              className="w-100 rounded"
              src="http://placekitten.com/g/480/320"
            />
          </noscript>
        </div>
      </div>
    </div>
  </div>,
]
`;
